uniffi_bindgen/src/bindings/gecko/mod.rs

```diff
@@ -0,0 +1,151 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use anyhow::{anyhow, bail, Context, Result};
+use std::{
+    ffi::OsString,
+    fs::File,
+    io::Write,
+    path::{Path, PathBuf},
+    process::Command,
+};
+
+pub mod gen_gecko;
+pub use gen_gecko::{
+    Config, Interface, InterfaceHeader, Namespace, NamespaceHeader, SharedHeader, WebIdl,
+};
+
+use super::super::interface::ComponentInterface;
+
+pub struct Source {
+    name: String,
+    header: String,
+    source: String,
+}
+
+pub struct Bindings {
+    webidl: String,
+    shared_header: String,
+    sources: Vec<Source>,
+}
+
+/// Generate uniffi component bindings for Gecko.
+///
+/// Bindings to a Rust interface for Gecko involves more than just generating a
+/// `.cpp` file.
```

Comment:
Let’s move the comment about `Codegen.py` out of `generate_bindings` and up here.